src/test/functional/pages/PimPage.ts

```diff
@@ -1,61 +1,182 @@
+      });    
+    }
+
+  public getConfirmDeleteButton(
+    ): Locator {
+      return  this.page.getByRole("button", {
+        name: Labels.YES_DELETE,
+      });
+    }
+
+  public getMultiplyDeleteButton(
+    ): Locator {
+      return  this.page.getByRole("button", {
+        name: Labels.DELETE_SELECTED,
+      });
+    }
+  
+  protected getEmployeeNameInput(
+    ): Locator {
+      return  this.page
+      .getByPlaceholder(Labels.TYPE_FOR_HINTS)
+      .first();
+    }
+
+  protected getEmployeeIdInput(
+    ): Locator {
+      return  this.page.getByRole("textbox").nth(2);
+    }
+
+  protected getSearchEmployeeButton(
+    ): Locator {
+      return  this.page.getByRole("button", {
+        name: Labels.SEARCH,
+      });
+    }
+
+  public getResetButton(
+    ): Locator {
+      return  this.page.getByRole("button", {
+        name: Labels.RESET,
+      });
+    }
+
+  // public getLocatorByRandomNewEmplyeeName(randomNewEmpoyeeName: string):Locator {
+  //   return this.page.locator(`.oxd-table-cell:has(:text("${randomNewEmpoyeeName}"))`);
+  // } this also works, let's keep it in codebase for poor times
   
-    public async searchEmployeeByName(employeeName:string) {
-      await this.employeeNameInput.type(employeeName)
-      await this.searchEmployeeButton.click()
+  public getCellByRandomNewEmplyeeName(
+    randomNewEmployeeName: string
+  ): Locator {
+    return this.page
+    .getByRole('cell', {name: randomNewEmployeeName})
+  }
+
+  // public getTrashBinByRandomEmployeeName(randomNewEmployeeName: string):Locator {
+  //   return this.page.locator(`.oxd-table-card:has(:text("${randomNewEmployeeName}")) .oxd-icon.bi-trash`);
+  // } 
+  // this also works, let's keep it in codebase for poor times
+
+  public getTrashBinByRandomEmployeeName(
+    randomNewEmployeeName: string
+  ): Locator {
+    return this.page
+      .getByText(randomNewEmployeeName)
+      .locator("xpath=../..")
+      .getByRole("button")
+      .locator(".bi-trash");
+  }
+
+  // public  getEditIconByRandomEmployeeName(randomNewEmpoyeeName: string): Locator {
+  //   return this.page.locator( `.oxd-table-card:has(:text("${randomNewEmpoyeeName}")) .oxd-icon.bi-pencil-fill`);
+  // } 
+  //this also works, let's keep it in codebase for poor times
+
+  public getEditIconByRandomEmployeeName(randomNewEmployeeName: string): Locator {
+    return this.page
+    .getByText(randomNewEmployeeName)
+    .locator('xpath=../..')
+    .getByRole('button')
+    .locator('.bi-pencil-fill')
+  }
+
+  // public  getCheckIconByRandomEmployeeName(randomNewEmpoyeeName: string):Locator {
+  //   return this.page.locator(`.oxd-table-card:has(:text("${randomNewEmpoyeeName}")) .oxd-checkbox-input-icon`);
+  // } 
+  // this also works, let's keep it in codebase for poor times
+
+  public getCheckIconByRandomEmployeeName(
+    randomNewEmployeeName: string
+  ): Locator {
+    return this.page
+      .getByText(randomNewEmployeeName)
+      .locator("xpath=../..")
```

Comment:
How does this locator work?

Reply:
In this particular case, ../.. is an XPath expression used to select the parent of the parent element. It's essentially asking Playwright to locate an element's grandparent. This could be useful when you want to select an element relative to its grandparent in the HTML structure.